compile/compile_test.go

```diff
@@ -1641,6 +1648,194 @@ update {
 	}
 }
 
+func TestCompilerOptimizationSupportRegoVersion(t *testing.T) {
+	tests := []struct {
+		note                string
+		modulesRegoVersion  ast.RegoVersion
+		regoV1ImportCapable bool
+		entrypoint          string
+		files               map[string]string
+		expected            []string
+	}{
+		{
+			note:                "v0 module, rego.v1 capable",
+			modulesRegoVersion:  ast.RegoV0,
+			regoV1ImportCapable: true,
+			entrypoint:          "test/p",
+			files: map[string]string{
+				"test.rego": `package test
+p {
+    input.x == 1
+}`,
+			},
+			expected: []string{
+				`package test
+
+import rego.v1
+
+p if {
+	input.x = 1
+}
+`,
+			},
+		},
+		{
+			note:                "v0 module, rego.v1 capable, rule name conflict with keyword",
+			modulesRegoVersion:  ast.RegoV0,
+			regoV1ImportCapable: true,
+			entrypoint:          "test/contains",
+			files: map[string]string{
+				"test.rego": `package test
+contains {
+    input.x == 1
+}`,
+			},
+			// rego.v1 import not used, since rule name conflicts with future keyword
```

Comment:
Maybe I'm missing something but if we have `regoV1ImportCapable: true`, then shouldn't there be a compile error?